### DISCUSSION TITLE
Include service and method info in HandlerContext (connect-node)

### DISCUSSION THREAD

**Discussion by fubhy** [connectrpc/connect-es#346] [Ideas]
Including the service and method info in the handler context would help with building generic functionality in and around individual method implementations.

For instance, this would allow generic middlewares like a cache wrapper or access control to be wrapped around method implementations generically.

I'm assuming that there'll be a notion of middleware / interceptor added for connect-node at some point? Even with that in mind I think it would be good to forward these values to the handler.

**Comment by timostamm** [comment]
Yes, adding service and method info to the context is a good call. If you'd like to have a go at it, we'd be happy to accept a contribution!

**Comment by timostamm** [comment]
For some thoughts in middleware, please also see https://github.com/bufbuild/connect-web/discussions/348#discussioncomment-4264171
